main.rb

```diff
@@ -0,0 +1,158 @@
+require './person'
+require './book'
+require './rental'
+require './classroom'
+require './student'
+require './teacher'
+
+# App class
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @choices = [
+      'List all books', 'List all people', 'Create a person', 'Create a book',
+      'Create a rental', 'List all rentals for a given person id', 'Exit'
+    ]
+  end
+
+  attr_reader :choices
+
+  def list_books_or_people(input)
+    case input
+    when 1
+      list_books
+    when 2
+      list_people
+    end
+  end
+
+  def list_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each_with_index { |person, i| puts "#{i}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person_or_create_book(input)
+    case input
+    when 3
+      create_person
+    when 4
+      create_book
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or teacher (2)? [Input the number]:'
+    user_option = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    if user_option == '1'
+      print 'Has parent permission? [Y/N]'
+      parent_permission = gets.chomp.downcase
+      parent_permission = parent_permission == 'y'
+
+      classroom = Classroom.new('Class A')
+
+      new_student = Student.new(age, name, classroom, parent_permission: parent_permission)
+      @people << new_student
+    else
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      new_teacher = Teacher.new(specialization, name, age)
+      @people << new_teacher
+    end
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] Upon creating a student, I realized that 2 or more students can have the same ID and that's because you set a person's age as their ID which is a bad idea as no 2 persons should have the same ID
- [ ] I created a teacher and tried to list all persons but it appears that name appears as age and vice versa. Kindly fix that.

![image](https://user-images.githubusercontent.com/16058470/151484992-917b2d7a-b640-40c6-abf5-7e54a5ee3cb9.png)

Reply:
Thank you for noticing this. It's been resolved.